### PR TITLE
Improve readability for test outputs

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
So far we have used `ObjectGoPrintSideBySide` in many places, which describes itself as

> // ObjectGoPrintSideBySide prints a and b as textual dumps side by side,
> // enabling easy visual scanning for mismatches.

I politely disagree with that statement. This is how `ObjectGoPrintSideBySide` prints the diff between 2 KKP Users, see how long it takes you to "easily scan for mismatches":

screenshot-2022-07-05-145005

I propose that we use YAML-based diffs, which are usually much easier to read. Here's the same diff, but as YAML:

screenshot-2022-07-05-145031

Especially since we in 99% of cases only care about _semantic_ equivalence, letting `ObjectGoPrintSideBySide` tell us about memory addresses is totally pointless.

This PR therefore adds a new `pkg/test/diff` package with a few helpers and then refactors our codebase to use those helpers wherever applicable.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/retest

vSphere seems stable now

File: pkg/test/diff/diff.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
```
Comment (ahmedwaleedmalik) [inline pkg/test/diff/diff.go]: nit

```suggestion
Copyright 2022 The Kubermatic Kubernetes Platform contributors.
```

**Comment by ahmedwaleedmalik** [comment]
Looks good, just one minor nit.

**Comment by xrstf** [comment]
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24-customfolder
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24-datastore-cluster
/hold cancel
/retest

**Comment by xrstf** [comment]
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24-customfolder
/override pre-kubermatic-e2e-vsphere-ubuntu-1.24-datastore-cluster
/override pre-kubermatic-ccm-migration-vsphere-e2e

**Comment by ahmedwaleedmalik** [comment]
/lgtm 
;)

**Pull request merged**